webbpsf/optics.py

```diff
@@ -1129,26 +1151,51 @@ def __init__(self, instrument, include_oversize=False, **kwargs):
 
                 # Cubic interpolation of all points
                 zgrid = griddata((v2, v3), zvals, (X, Y), method='cubic')
-    
+
+                # Want to rotate zgrid image of some SIs to minimize NaN clipping
+                if 'NIRSpec' in lookup_name:
+                    rot_ang = 42
+                elif 'MIRI' in lookup_name:
+                    rot_ang = -25
```

Comment:
Where does a rotation angle of -25 deg for MIRI come from? The MIRI detector axes are only rotated ~ 4.5 deg relative to the V2V3 frame. Likewise I'm surprised to see a 5 deg rotation for NIRISS, which is less than a degree rotated relative to V2V3.

Reply:
It's more about the shape of the polygons of the measured field points rather than the detector orientation. These rotation angles maximize the number of remaining field points after trimming NaNs on the edges.

Reply:
For instance, have a look at the level of skewness for the MIRI WFE map:

![image](https://user-images.githubusercontent.com/17557291/55829018-88fdcb00-5ac2-11e9-8f6a-c3f6b3abc7fb.png)

Reply:
Ok, I decided to look into developing a more optimized NaN trimming algorithm. The new algorithm does a much better job keeping real data rather than over-trimming. Optimal rotation angles are much closer to the real instrument rotation relative to V2V3.

Reply:
Here's a final visualization. You'll probably want to disable SI WFE measurements for MIRI coronagraphy given there are no measurements in that part of the field, and RGI simply spits out artifacts.

![Figure_6](https://user-images.githubusercontent.com/17557291/55844225-1d7f2200-5af1-11e9-90d7-a91e1a585dd4.png)

Reply:
Thanks for continuing to work on improving this!  I'll take a look and test it a bit sometime when I'm not in the middle of a wavefront rehearsal in the MOC. 

Can you share your code or notebook for making these plots, please? Could be nice to include some version of these in the WebbPSF documentation. (That doesn't necessarily need to be in the repo - you could just email me that if it's easier, or you could PR it into the dev_utils subdir. However you prefer.)

Reply:
Uploaded a notebook for making the plots. Also included a couple minor improvements that reduces artifacts on extrapolated regions.